sql/mdl.cc

```diff
@@ -3394,8 +3394,9 @@ bool MDL_lock::object_lock_kill_conflicting_locks(MDL_context *ctx,
 
 bool MDL_context::acquire_lock_nsec(MDL_request *mdl_request,
                                     ulonglong lock_wait_timeout_nsec) {
-  THD *thd = get_thd();
   bool is_high_priority_ddl = false;
+#if !defined(EXTRA_CODE_FOR_UNIT_TESTING)
```

Comment:
This looks like a hack to me: this definition should be used to add extra code to testing, not to remove some functionality. It just hides the real issue - why we get those undefined externals.

Reply:
`merge_small_tests` at  `unittest/gunit/CMakeLists.txt` links only:
```
SET(SQL_GUNIT_LIB_SOURCE 
  ${CMAKE_SOURCE_DIR}/sql/filesort_utils.cc 
  ${CMAKE_SOURCE_DIR}/sql/mdl.cc
  ${CMAKE_SOURCE_DIR}/sql/sql_list.cc
  ${CMAKE_SOURCE_DIR}/sql/stateless_allocator.cc
  ${CMAKE_SOURCE_DIR}/sql-common/sql_string.cc
  ${CMAKE_SOURCE_DIR}/sql/thr_malloc.cc
  )
```
These missing functions are defined in `sql_parse.cc`. When I added `sql_parse.cc` to `SQL_GUNIT_LIB_SOURCE` there is a lot more `error: undefined reference to ...`.

Reply:
There are still other choices:

* these unresolved symbols don't really depend on anything, it would be possible to extract them to somewhere else, which can be added to the unit tests
* or the unit tests already use dummy implementations of some functions/variables, those could also be extended with these

Reply:
Solution 1 requires to introduce a new file or move undefined functions to `mdl.cc`.
Solution 2 works:```
--- a/sql/mdl.cc
+++ b/sql/mdl.cc
@@ -4971,3 +4971,14 @@ String timeout_message(const char *command, const char *name1,
   }
   return msg;
 }
+
+
+#if defined(EXTRA_CODE_FOR_UNIT_TESTING)
+
+bool slave_high_priority_ddl = 0;
+ulonglong slave_high_priority_lock_wait_timeout_nsec = 1.0;
+
+bool support_high_priority(enum enum_sql_command) noexcept { return false; };
+bool Security_context::check_access(ulong, bool) { return false; };
+
+#endif
```
but I'm not convinced it's a better solution because it "hides" that high priority DDL are disabled for unit tests.